text/0061-defrag-memos.md

```diff
@@ -0,0 +1,71 @@
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+Since defragmentation transactions never have change, they could only produce a single output. A second output is not necessary for RTH because the sender and recipient of defragmentation transactions are the same. This proposal does not remove the change `TxOut` from defragmentation transactions as it would make defragmentation transactions easier to identify. Keeping two outputs per transaction makes all transactions appear the same.
+
+Since it is proposed that defragmentation transactions will produce two outputs, the fee and outlay fields of the change output memo are set to 0. Rather than simply writing the same exact data to both memos, the change output is given 0 fee and outlay so that clients can add up the fees and outlays of defragmentation `TxOut`s with the same ID. If the same memo data was written to both outputs, fees and outlays would be double if added up. To produce the correct data, clients would need to check the `Amount` on each `TxOut` and choose to ignore the defragmentation memos on `TxOut`s` with a value of 0. This raises the knowledge barrier needed for developers to utilize the defragmentation memos. Setting the change output memo fees and outlays to 0 allows them to be added to the regular outputs as it is an identity operation. Whether or not clients manually ignore change outputs, the math will have the same result.
+
+One alternative to defragmentation memos would be to continue using RTH for identifying defragmentation transactions. But this approach has a higher barrier of entry for client developers in terms of MobileCoin knowledge. It is much simpler to just check the memo type; and when it comes to sharing example code, defragmentation memos are much easier to demonstrate.
+
+Including a defragmentation ID may also not be necessary. Conversely, it may be better to use a larger value. A single u64 was chosen for simplicity and should be enough for one account. If clients do not need the ID field, they can also choose to ignore it.
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+- ~~Is a defragmentation ID necessary or valuable? Should it be larger than 64 bits?~~
+- ~~Does this memo need any sort of validation? If client applications are hiding defragmentation transactions, users with custom clients could create 'invisible' transactions. This doesn't seem like a major issue, however.~~
+
+# Future possibilities
+[future-possibilities]: #future-possibilities
+
+This proposal should make it much easier for developers to create apps that behave more like regular payment apps. It will also reduce the amount of code needed to identify defragmentation transactions.
```

Comment:
:wrench: This feels like "rational" not "future possibilities" suggest moving this into the rational portion

Reply:
I think you're right, and these points are covered by Motivation already, so I'm fine to remove these.